src/stdio/printf.c

```diff
@@ -288,7 +288,7 @@ static size_t _ftoa(out_fct_type out, char* buffer, size_t idx, size_t maxlen, d
   double diff = 0.0;
 
   // if input is larger than thres_max, revert to exponential
-  const double thres_max = (double)0x7FFFFFFF;
+  const double thres_max = (double)0x7FFFFFFFFFFFFFFF;
```

Comment:
sign (1) + 19 digits + "." (1) + 9 precision = 30, which is barely safe (PRINTF_FTOA_BUFFER_SIZE == 32)

Reply:
I added a protection to make the maximum precision at 9, so the longest result will be `-9223372036854775808.000000000` which is 30 characters, still safe.